projects/plugins/jetpack/modules/google-fonts.php

```diff
@@ -102,6 +102,201 @@ function jetpack_add_google_fonts_provider() {
+
+	return $global_styles;
+}
+
+/**
+ * Updates the global styles CPT.
+ *
+ * @param array  $new_settings New global styles to update.
+ * @param array  $new_styles New global styles settings to update.
+ * @param int    $user_custom_post_type_id ID of global styles CPT.
+ * @param object $global_styles_controller Controller that handles REST requests for global styles.
+ *
+ * @return void
+ */
+function update_global_styles( $new_settings, $new_styles, $user_custom_post_type_id, $global_styles_controller ) {
+	$update_request = new WP_REST_Request( 'PUT', '/wp/v2/global-styles/' );
+	$update_request->set_param( 'id', $user_custom_post_type_id );
+	$update_request->set_param( 'settings', $new_settings );
+	$update_request->set_param( 'styles', $new_styles );
+
+	$global_styles_controller->update_item( $update_request );
+	delete_transient( 'global_styles' );
+	delete_transient( 'global_styles_' . get_stylesheet() );
+	delete_transient( 'gutenberg_global_styles' );
+	delete_transient( 'gutenberg_global_styles_' . get_stylesheet() );
+}
+
+/**
+ * Undoes changes made to custom fonts in the site editor global styles so that blockbase
+ * custom fonts can behave as expected.
+ *
+ * @return void
+ */
+function revert_migrated_blockbase_custom_fonts() {
+	$user_custom_post_type_id = WP_Theme_JSON_Resolver_Gutenberg::get_user_global_styles_post_id();
+	$global_styles_controller = new Gutenberg_REST_Global_Styles_Controller();
+
+	$global_styles = fetch_global_styles( $user_custom_post_type_id, $global_styles_controller );
+
+	// converts data to array (in some cases settings and styles are objects insted of arrays)
+	$new_settings = (array) $global_styles->data['settings'];
+	$new_styles   = (array) $global_styles->data['styles'];
+
+	if ( isset( $new_styles['typography']['fontFamily'] ) ) {
+		unset( $new_styles['typography']['fontFamily'] );
+	}
+
+	// TODO: Unset post title and headings block typography
+
+	$new_settings = array_merge(
+		$new_settings,
+		array(
+			'typography' => array(
+				'fontFamilies' => get_option( 'blockbase_legacy_font_settings' ),
+			),
+		)
+	);
+
+	update_global_styles( $new_settings, $new_styles, $user_custom_post_type_id, $global_styles_controller );
+	delete_option( 'blockbase_legacy_font_settings' );
+}
+
+/**
+ * Transforms data saved by the blockbase fonts customizer feature to make things
+ * compatible with custom fonts for global styles in the site editor
+ *
+ * @return void
+ */
+function migrate_blockbase_custom_fonts() {
+	// The data has already been transformed
+	if ( get_option( 'blockbase_legacy_font_settings' ) ) {
+		// Gutenberg or Jetpack versions have changed or the plugins have been uninstalled. If custom google
+		// fonts are no longer supported in the site editor, revert our transformed data so that the original
+		// blockbase fonts customizer feature can behave properly.
+		if ( ! blockbase_supports_jetpack_google_fonts() ) {
+			revert_migrated_blockbase_custom_fonts();
```

Comment:
**TODO:**

Add other conditions when this is called. I imagine we'll want to hook into actions that run when the google fonts jetpack module is deactivated to execute this reversion logic.